pkg/server/container_list_test.go

```diff
@@ -209,56 +225,145 @@ func TestListContainers(t *testing.T) {
 		},
 		{
 			metadata: containerstore.Metadata{
-				ID:        "4",
+				ID:        "c-4container",
 				Name:      "name-4",
-				SandboxID: "s-2",
+				SandboxID: "s-2abcdef1234",
 				Config:    &runtime.ContainerConfig{Metadata: &runtime.ContainerMetadata{Name: "name-4"}},
 			},
 			status: containerstore.Status{
 				CreatedAt: createdAt,
 			},
 		},
 	}
-	filter := &runtime.ContainerFilter{
-		PodSandboxId: "s-1",
-	}
-	expect := []*runtime.Container{
-		{
-			Id:           "1",
-			PodSandboxId: "s-1",
-			Metadata:     &runtime.ContainerMetadata{Name: "name-1"},
-			State:        runtime.ContainerState_CONTAINER_CREATED,
-			CreatedAt:    createdAt,
-		},
-		{
-			Id:           "2",
-			PodSandboxId: "s-1",
-			Metadata:     &runtime.ContainerMetadata{Name: "name-2"},
-			State:        runtime.ContainerState_CONTAINER_RUNNING,
-			CreatedAt:    createdAt,
-		},
-		{
-			Id:           "3",
-			PodSandboxId: "s-1",
-			Metadata:     &runtime.ContainerMetadata{Name: "name-3"},
-			State:        runtime.ContainerState_CONTAINER_EXITED,
-			CreatedAt:    createdAt,
-		},
+
+	// Inject test sandbox metadata
+	for _, sb := range sandboxesInStore {
+		assert.NoError(t, c.sandboxStore.Add(sb))
 	}
 
-	// Inject test metadata
+	// Inject test container metadata
 	for _, cntr := range containersInStore {
 		container, err := cntr.toContainer()
 		assert.NoError(t, err)
 		assert.NoError(t, c.containerStore.Add(container))
 	}
 
-	resp, err := c.ListContainers(context.Background(), &runtime.ListContainersRequest{Filter: filter})
-	assert.NoError(t, err)
-	require.NotNil(t, resp)
-	containers := resp.GetContainers()
-	assert.Len(t, containers, len(expect))
-	for _, cntr := range expect {
-		assert.Contains(t, containers, cntr)
+	for testdesc, testdata := range map[string]struct {
+		filter *runtime.ContainerFilter
+		expect []*runtime.Container
+	}{
+		"test without filter": {
+			filter: &runtime.ContainerFilter{},
```

Comment:
>After this refactor, I prefer put add 3 more test cases for trunc index, instead of having an inner loop.

Do you mean this loop below?
		"for _, isTrunc := range []bool{false, true} {"

It can save some code.

Reply:
I mean after you have a predefined expected container list, it won't save too much code.
It seems more clear to make them separate test cases.

Reply:
done